### PR TITLE
Add castArray

### PR DESCRIPTION
**Pull request by robinpokorny**
Add `castArray` for ensuring array.

I was not sure about returning a copy, but I cannot see how to stick to immutability otherwise.

### CODE REVIEW

File: documentation/README.md [hunk 1]
```diff
@@ -177,7 +178,7 @@ const assign = require('1-liners/assign');
 
 const yedi = { id: 1, age: 100 };
 
-assign({ name: 'Yoda', age: 900 }, yedi);  // => { id: 1, age: 900, name: 'Yoda' }
+assign({ name: 'Yoda', age: 900 }, yedi);  // => { id: 1, name: 'Yoda', 900 }
```
Comment (hemanth) [inline documentation/README.md]: ?

Comment (robinpokorny) [inline documentation/README.md]: The file was automatically generated.
This is in the master: https://github.com/1-liners/1-liners/blob/master/module/assign.js#L14

Should I fix it in this PR?

Comment (hemanth) [inline documentation/README.md]: Not to worry, I shall fix it, thanks.

File: module/castArray.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ * @example
+ *
+ * 	const castArray = require('1-liners/castArray');
+ *
+ * 	castArray([1, 2, 3]);  // => [1, 2, 3]
+ * 	castArray(1);  				 // => [1]
+ * 	castArray(null);  	   // => [null]
+ *
+ */
+export default (value) => Array.isArray(value) ? value.slice(0) : [value];
```
Comment (stoeffel) [inline module/castArray.js]: instead of `slice` we could do `[...value]`

Comment (robinpokorny) [inline module/castArray.js]: @stoeffel, I've read that `.slice(0)` is the fastest. `[...value]` is currently transpiled to a weird thing, see https://babeljs.io/repl/#?evaluate=true&presets=es2015&experimental=true&loose=false&spec=true&code=%5B...value%5D.

On the other hand it is probably more readable.

**Pull request merged**